src/utils/api.ts

```diff
@@ -432,10 +433,61 @@ class Api {
       resource: this.barSearchUrl,
     });
 
-  getFishTypes = async (): Promise<{ id: string; label: string; photo: any }[]> =>
+  getFishTypes = async (): Promise<FishType[]> =>
     await this.getAll({
       resource: Resources.FISH_TYPES,
     });
+
+  getResearches = async (): Promise<GetAllResponse<Research>> =>
+    await this.get({
+      resource: `researches`,
+      populate: ['user'],
+    });
+
+  createResearch = async (params: any): Promise<Research> =>
+    await this.post({
+      resource: `researches`,
+      params,
+    });
+
+  updateResearch = async (params: any, id: string): Promise<Research> =>
+    await this.patch({
+      resource: `researches`,
+      params,
+      id,
+    });
+
+  getResearch = async (id: string): Promise<Research> =>
+    await this.getOne({
+      resource: `researches`,
+      id,
+      populate: ['fishes', 'geom'],
+    });
+
+  uploadFiles = async (files: File[] = []): Promise<any> => {
+    if (isEmpty(files)) return [];
```

Comment:
Ar tau cia tikrai reikia lodasha naudot? Butu gerai jo isvis nenaudoti jeigu galim ir isimti is projekto.